WooCommerce/Classes/Blaze/BlazeLocalNotificationScheduler.swift

```diff
@@ -0,0 +1,166 @@
+import Yosemite
+import protocol Storage.StorageManagerType
+import Combine
+
+protocol BlazeLocalNotificationScheduler {
+    func scheduleNotifications() async
+}
+
+/// Handles the scheduling of Blaze local notifications.
+///
+final class DefaultBlazeLocalNotificationScheduler: BlazeLocalNotificationScheduler {
+    private let siteID: Int64
+    private let stores: StoresManager
+    private let storageManager: StorageManagerType
+    private let scheduler: LocalNotificationScheduler
+    private let userDefaults: UserDefaults
+    private let pushNotesManager: PushNotesManager
+    private var subscriptions: Set<AnyCancellable> = []
+    private let blazeEligibilityChecker: BlazeEligibilityCheckerProtocol
+
+    /// Blaze campaign ResultsController.
+    private lazy var blazeCampaignResultsController: ResultsController<StorageBlazeCampaignListItem> = {
+        let predicate = NSPredicate(format: "siteID == %lld", siteID)
+        let sortDescriptorByStartTime = NSSortDescriptor(keyPath: \StorageBlazeCampaignListItem.startTime,
+                                                         ascending: false)
+        let resultsController = ResultsController<StorageBlazeCampaignListItem>(storageManager: storageManager,
+                                                                                matching: predicate,
+                                                                                sortedBy: [sortDescriptorByStartTime])
+        return resultsController
+    }()
+
+    init(siteID: Int64,
+         stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         userDefaults: UserDefaults = .standard,
+         pushNotesManager: PushNotesManager = ServiceLocator.pushNotesManager,
+         blazeEligibilityChecker: BlazeEligibilityCheckerProtocol = BlazeEligibilityChecker()) {
+        self.siteID = siteID
+        self.stores = stores
+        self.storageManager = storageManager
+        self.pushNotesManager = pushNotesManager
+        self.scheduler = LocalNotificationScheduler(pushNotesManager: pushNotesManager)
+        self.userDefaults = userDefaults
+        self.blazeEligibilityChecker = blazeEligibilityChecker
+    }
+
+    /// Starts observing campaigns from storage and schedules local notification
+    ///
+    func scheduleNotifications() async {
+        guard await isEligibleForBlaze() else {
+            DDLogDebug("Blaze: Store not eligible for Blaze. Don't schedule local notification.")
+            return
+        }
```

Comment:
Suggestion: We should also cancel any local notifications related to Blaze just in case some were scheduled before the site becomes ineligible. This is mostly to be safe but worth the effort I believe.

Reply:
Good suggestion. Canceled notification and added unit test in https://github.com/woocommerce/woocommerce-ios/pull/13816/commits/e189a3ff70e048a474f9137aa0c47b08369ced33